pandas/core/arrays/base.py

```diff
@@ -1156,9 +1179,9 @@ def _maybe_convert(arr):
                     # https://github.com/pandas-dev/pandas/issues/22850
                     # We catch all regular exceptions here, and fall back
                     # to an ndarray.
-                    try:
-                        res = self._from_sequence(arr)
-                    except Exception:
+                    res = try_cast_to_ea(self, arr)
```

Comment:
The documentation says the first parameter is an `ExtensionArray subclass`.  I don't think this counts?

Reply:
We're calling a classmethod of the EA subclass, so it works with an instance too.  I don't think this benefits much from changing `self` to `type(self)`, but can do if you have a strong opinion.

Reply:
I have no problem with the implementation, but the documentation should be clarified to indicate that we're accepting both the class itself and instances.

I would consider changing the parameter name as well, since `cls` is often used to indicate the class itself (and not instances).  Maybe call it `cls_or_instance` ?